src/utils/decodeActorCID.ts

```
@@ -1,14 +1,14 @@
 import base32Decode from 'base32-decode'
 
-export function decodeActorCID(cid: string): string {
+export const decodeActorCID = (cid: string): string => {
   try {
     const decoded = base32Decode(cid.slice(1).toUpperCase(), 'RFC4648')
     const actorCode = new TextDecoder('utf-8').decode(decoded.slice(4))
     if (!actorCode.includes('fil/')) {
-      return 'Unknown actor'
+      throw new Error('unknown actor code')
     }
     return actorCode
   } catch {
-    return 'Unknown actor'
+    throw new Error('invalid actor code')
```

@Schwartz10 This is basically the way the util was implemented in the Safe.

It seemed to make sense to use the `react-components` implementation in the Safe, but probably better to throw an error so that invalid actors can be caught instead of just displaying `Unknown actor`. Not sure how the implementation of custom actors will affect all this, but I guess we will cross that bridge when we get there